notebooks/experiment_storage_penetration.jl

```diff
@@ -0,0 +1,329 @@
+	using Convex, ECOS
+	using Plots
+	using PlutoUI
+	using JLD
+	using LinearAlgebra
+	
+	using Revise
+	using CarbonNetworks
+	
+	OPT = () -> ECOS.Optimizer(verbose=false)
+end;
+
+# ╔═╡ 257a6f74-d3c3-42eb-8076-80d26cf164ca
+theme(:default, label=nothing, 
+		tickfont=(:Times, 8), guidefont=(:Times, 8), legendfont=(:Times, 8))
+
+# ╔═╡ 9bd515d4-c7aa-4a3d-a4fb-28686290a134
+md"""
+## Generate data
+"""
+
+# ╔═╡ 75dfaefd-abec-47e2-acc3-c0ff3a01048e
+md"""
+### Network
+"""
+
+# ╔═╡ f999d732-14b3-4ac5-b803-3df7a96ef898
+begin
+	net, d_peak = load_synthetic_network("case30.m")
+	n = length(d_peak)
+	
+	# Limit line flows and generation capacities
+	net.pmax *= 0.75
+	net.gmax *= 0.75
+	
+	# Remove lines
+	# net.pmax[[2, 3, 5, 11, 12, 13, 14, 20, 22, 26, 27, 30, 31, 33, 39]] .= 0	
+	net.pmax[[2, 3, 5, 11, 12, 20, 22, 26, 27, 30, 39]] .= 0
+	
+	"Network loaded."
+end
+
+# ╔═╡ 23690382-3d30-46e3-b26a-a30875be78ec
+begin
+	using LightGraphs, SimpleWeightedGraphs, GraphPlot
+	Random.seed!(5)
+	
+	G = SimpleWeightedGraph(n)
+	for j in 1:size(net.A, 2)
+		inds = findall(x -> x != 0, net.A[:, j])
+		add_edge!(G, inds[1], inds[2], net.pmax[j])
+	end
+	
+	plt = gplot(G)
+end
+
+# ╔═╡ 1bd72281-4a7f-44f4-974d-632e9d0aaf28
+md"""
+### Demand and renewable time series
+"""
+
+# ╔═╡ 0c786da1-7f44-40af-b6d6-e0d6db2242b2
+demand_data = load_demand_data("2021_07_01", normalize_rows=true);
+
+# ╔═╡ 5b80ca83-0719-437f-9e51-38f2bed02fb4
+renew_data, renew_labels = load_renewable_data("2021_07_01"; normalize_rows=true);
+
+# ╔═╡ 34d4bd62-6be2-4089-8caa-1a8715bee433
+md"""
+### Map time series to network data
+"""
+
+# ╔═╡ b7476391-30b9-4817-babf-7c9078531ee7
+T, n_demand = size(demand_data)
+
+# ╔═╡ cfcba5ad-e516-4223-860e-b1f18a6449ba
+begin
+	plt1 = plot(demand_data[:, rand(1:n_demand, 5)], lw=2, palette=:Blues)
+	plt2 = plot(renew_data, lw=2, palette=:Reds)
+	
+	plot(plt1, plt2, layout=(2, 1), size=(600, 300))
+end
+
+# ╔═╡ c82ef027-740a-49b1-93d2-1554c411a896
+renewable_pentration = 0.5
+
+# ╔═╡ 0239e1da-caf5-4593-af1b-5d1e8d2f2b3e
+begin
+	Random.seed!(1)
+	demand_profiles = rand(1:n_demand, n)
+end;
+
+# ╔═╡ c75d996b-0847-4cc7-bc93-3d42db39f1bd
+n_renew = size(renew_data, 2)
+
+# ╔═╡ 522e95d5-15a8-47ec-a79c-c4cc17cf86fd
+begin
+	Random.seed!(2)
+	renew_profiles = rand(0:n_renew, n)
+	
+	
+	get_renew = (t, renew_profile) -> 
+		renew_profile == 0 ? 0.0 : renew_data[t, renew_profile]
+	
+	d_dyn = [
+		d_peak .* (demand_data[t, demand_profiles] - renewable_pentration * get_renew.(t, renew_profiles))
+		for t in 1:T
+	]
+end;
+
+# ╔═╡ bfa4a8f9-cfcd-4e22-b2dc-751226f3a73c
+begin
+	Random.seed!(4)
+	plot(size=(600, 200))
+	plot!(hcat(d_dyn...)'[:, rand(1:n, 10)], lw=2, c=repeat(renew_profiles, outer=T)')
+end
+
+# ╔═╡ a8ccbc8e-24e6-4214-a179-4edf3cf26dad
+md"""
+### Carbon emissions data
+"""
+
+# ╔═╡ 897cc3a0-66cb-417d-81d5-20036b32dd99
+net.gmax
+
+# ╔═╡ e8ee5cbb-4afc-4737-b006-90071f6138cd
+begin
+	Random.seed!(3)
+	emission_rates = 1000 .+ 1000*rand(length(net.gmax))
```

Comment:
Are these emission rates the same as the carbon costs? Or do they represent something else?